packages/lit-html/rollup.config.js

```diff
@@ -13,36 +13,108 @@
+// 2) Don't include copyright header comments.
+// 3) Don't include the "//# sourceMappingURL" comment.
+const CHECKSIZE = !!process.env.CHECKSIZE;
+if (CHECKSIZE) {
+  console.log('NOTE: In CHECKSIZE mode, no output!');
+}
+
+const skipBundleOutput = {
+  generateBundle(options, bundles) {
+    // Deleting all bundles from this object prevents them from being written,
+    // see https://rollupjs.org/guide/en/#generatebundle.
+    for (const name in bundles) {
+      delete bundles[name];
+    }
+  },
+};
+
+const entryPoints = ['lit-html', 'directives/if-defined'];
 
 export default {
-  input: 'lib/lit-html.js',
+  input: entryPoints.map((name) => `development/${name}.js`),
   output: {
-    file: 'lit-html.bundled.js',
+    dir: './',
     format: 'esm',
+    // Preserve existing module structure (e.g. preserve the "directives/"
+    // directory).
+    preserveModules: true,
+    sourcemap: !CHECKSIZE,
   },
   plugins: [
+    // Switch all DEV_MODE variable assignment values to false. Terser's dead
+    // code removal will then remove any blocks that are conditioned on this
+    // variable.
+    //
+    // Code in our development/ directory looks like this:
+    //
+    //   const DEV_MODE = true;
+    //   if (DEV_MODE) { // dev mode stuff }
+    //
+    // Note we want the transformation to `goog.define` syntax for Closure
+    // Compiler to be trivial, and that would look something like this:
+    //
+    //   const DEV_MODE = goog.define('lit-html.DEV_MODE', false);
+    //
+    // We can't use terser's compress.global_defs option, because it won't
+    // replace the value of a variable that is already defined in scope (see
+    // https://github.com/terser/terser#conditional-compilation). It seems to be
+    // designed assuming that you are _always_ using terser to set the def one
+    // way or another, so it's difficult to define a default in the source code
+    // itself.
+    replace({
+      'const DEV_MODE = true': 'const DEV_MODE = false',
+    }),
+    // This plugin automatically composes the existing TypeScript -> raw JS
+    // sourcemap with the raw JS -> minified JS one that we're generating here.
+    sourcemaps(),
     terser({
       warnings: true,
       ecma: 2017,
       compress: {
         unsafe: true,
+        // An extra pass can squeeze out an extra byte or two.
+        passes: 2,
       },
       output: {
-        comments: false,
+        // "some" preserves @license and @preserve comments
+        comments: CHECKSIZE ? false : 'some',
         inline_script: false,
-        // TODO (justinfagnani): benchmark
-        // wrap_func_args: false,
       },
       mangle: {
         properties: {
           regex: /^__/,
+          // TODO A test in lit-html_test.ts calls
+          // `part.__element.dispatchEvent`, but when testing the production
+          // output, this will fail since `__element` is renamed. Is there
+          // another way to write this test, or should this property be public?
```

Comment:
We're going to make these public. You can do it in this PR

Reply:
Done (`__element` -> `element`)